File: c/src/pje_0007.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>

int main() {
	long primes[10001];
	primes[0] = 2;
	primes[1] = 3;
	primes[10000] = 0;
	long n = 3;
	int t = 2;
	while (primes[10000] == 0) {
		n += 2;
		int x;
		bool isPrime = true;
		long max = (int) (ceil(sqrt(n)));
		for (x = 0; x < t && primes[x] <= max; x++) {
			if (n%primes[x] == 0)
				isPrime = false;
		}
		if (isPrime) {
			primes[t] = n;
			t++;
		}
	}
	printf("%ld\n", primes[10000]);
	return 0;
}

File: python/src/pje_0007.py
primes = [2, 3]
n = 3
t = 2

while t < 10001:
	n = n + 2
	i = 0
	is_prime = 1
	while i < t and is_prime == 1:
		if n % primes[i] == 0: is_prime = 0
		i = i + 1
	if is_prime == 1:
		t = t + 1
		primes.append(n)

print "%d" % (primes[10000])
